web/modules/custom/dpl_library_agency/src/Branch/FallbackBranchRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\dpl_library_agency\Branch;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Repository with a fallback option.
+ *
+ * If an error/exception occurs when retrieving branches from one repository
+ * this will fall back to a secondary option.
+ */
+class FallbackBranchRepository implements BranchRepositoryInterface {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private BranchRepositoryInterface $primaryRepository,
+    private BranchRepositoryInterface $secondaryRepository,
+    private LoggerInterface $logger
+  ) {}
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getBranches(): array {
+    try {
+      return $this->primaryRepository->getBranches();
+    }
+    catch (\Exception $e) {
```

Comment:
It is unclear to me where this exception is being thrown. I did a bit of investigation, and perhaps I missed it. But a @throws in the docblock of the primaryRepository:getBranches() would be nice.

Reply:
As any branch repository implemenetation can be set as the primary I cannot see how I can set this in the fallback implementation.

I have updated the DocBlock in the FBS repository with a `@throws` block.